coderd/database/querier_test.go

```diff
@@ -1258,6 +1256,15 @@ func TestQueuePosition(t *testing.T) {
 		time.Sleep(time.Millisecond)
 	}
 
+	// Create default provisioner daemon:
+	dbgen.ProvisionerDaemon(t, db, database.ProvisionerDaemon{
+		Name:         "default_provisioner",
+		Provisioners: []database.ProvisionerType{database.ProvisionerTypeEcho},
+		// Ensure the `tags` field is NOT NULL for the default provisioner;
+		// otherwise, it won't be able to pick up any jobs.
+		Tags: database.StringMap{"a": "1"},
```

Comment:
Does `null` tags break things?

Reply:
`<@` - Checks if the left operand is contained within the right operand.

```SQL
SELECT NULL::jsonb <@ NULL::jsonb; -> returns NULL
SELECT NULL::jsonb <@ '{}'::jsonb; -> returns NULL
SELECT '{}'::jsonb <@ '{}'::jsonb; -> return true
```

so `provisioner_tagset_contains(pd.tags, pj.tags)` returns NULL if either of arguments is NULL

My assumption was that in production we don't have NULL tags, so we don't have to worry about it?
Also I think I can use `{}` instead of `{"a": "1"}`

Reply:
I see that we already use this approach in other places: https://github.com/coder/coder/blob/main/coderd/database/querier_test.go#L1250

Reply:
slightly improved it: https://github.com/coder/coder/pull/16685/commits/c1f421dc990c096d4f586e5c5082e4666db0bc32